### PR TITLE
Extend getupgrades endpoint for kubeone

### LINKED ISSUE #9409: Gets possible cluster upgrades for Kubeone cluster
### What would you like to be added? / User Story
Gets possible cluster upgrades for Kubeone cluster

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: Extend getupgrades endpoint of external cluster for kubeone

**Does this PR close any issues?**:
Fixes #9409

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v2/types.go [hunk 1]
```diff
@@ -398,30 +398,30 @@ type ExternalClusterState string
 
 const (
 	// PROVISIONING state indicates the cluster is being created.
-	PROVISIONING ExternalClusterState = "PROVISIONING"
+	PROVISIONING ExternalClusterState = "Provisioning"
```
Comment (zreigz) [inline pkg/api/v2/types.go:401]: it's not compatible with the UI. The frontend will stop working

Comment (imharshita) [reply pkg/api/v2/types.go:401]: will talk to sig-ui, if something can be done

File: go.mod [hunk 2]
```diff
@@ -82,7 +82,7 @@ require (
 	k8c.io/operating-system-manager v0.4.0
 	k8s.io/api v0.23.4
 	k8s.io/apiextensions-apiserver v0.23.4
-	k8s.io/apimachinery v0.23.4
+	k8s.io/apimachinery v0.23.5
```
Comment (xrstf) [inline go.mod:85]: Why update this one dependency?

Comment (imharshita) [reply go.mod:85]: no specific reason, i will change it back to what it was.

**Pull request merged**